### ISSUE TITLE
[BUG] salt-master in auto_accept, minion key appears in "Accepted Keys" and "Denied Keys"

### ISSUE THREAD

**Issue by arizvisa** [saltstack/salt#57648]
**Description**
I have a master configured with `auto_accept`. After using salt-cloud to deploy to a windows minion. The minion is first started in a "bootstrap" environment which is used to install some additional configuration such as upgrading the version of python and and configuring the etcd returner. Once that's complete, it switches to the "base" environment and reboots the machine. (edited) Upon failure it reboots and tries again..because windows (/edited)

Upon restarting the machine, it's unable to authenticate to the master. Looking at `salt-key` on the master, it seems that the same public key for the minion has now appeared in both the "Accepted Keys" and "Denied Keys" groups.

What situation would result in the key being categorized as a "Denied Keys"?

**Setup**
The master and bootstrap environment that's executed on the minion is the following, but it's very basic. salt-cloud is responsible for dropping the minion key (which appears correct on the minion and master)
https://github.com/arizvisa/lolfuzz3/blob/master/build/master.project/salt/remote-minion-windows.sls 
https://github.com/arizvisa/lolfuzz3/blob/master/build/master.project/salt/remote-minion-common.sls

**Steps to Reproduce the behavior**
Set the default environment to bootstrap, apply these states..reboot. (edited) I haven't done much research in this area, however. (/edited)

**Expected behavior**
As the keys haven't changed, one would expect the minion to still be able to authenticate. As the master is in auto-accept, the public key should be accepted.

**Screenshots**
![image](https://user-images.githubusercontent.com/2035497/84470669-755f6e80-ac49-11ea-8ba2-32cdb063c159.png)

**Versions Report**
Salt Version:
           Salt: 3000.3

Dependency Versions:
           cffi: 1.12.2
       cherrypy: 17.4.1
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: Not Installed
   pycryptodome: 3.8.1
         pygit2: Not Installed
         Python: 3.5.4 (v3.5.4:3f56838, Aug  8 2017, 02:07:06) [MSC v.1900 32 bi
t (Intel)]
   python-gnupg: 0.4.4
         PyYAML: 5.1.2
          PyZMQ: 18.0.1
          smmap: 2.0.5
        timelib: 0.2.4
        Tornado: 4.5.3
            ZMQ: 4.3.1

System Versions:
           dist:
         locale: cp1252
        machine: x86
        release: 8.1
         system: Windows
        version: 8.1 6.3.9600 SP0 Multiprocessor Free

(edited: cleaned up english a bit added extra comments)

**Comment by frogunder** [comment]
@saltstack/team-windows @twangboy Can you take a look at this Windows issue?

Thanks.

**Comment by arizvisa** [comment]
I'm pretty sure it's happening when I restart the salt-minion into a new environment. But I'm not 100% sure.

This capability is very key to the way I'm doing things because this is part of a project that automatically generates a number of salt-configured clusters. So, everything needs to be automatic \_and\_ platform-independent (hence the "bootstrap" environment being used to normalize things before transitioning into the "base" environment".

**Comment by damon-atkins** [comment]
Also note Salt-Minion does not handle sysprep.  I suspect you have a left over pki key on the minion.

**Comment by arizvisa** [comment]
What do you mean by handling sysprep exactly? If you're referring to the generalization stages of installing Windows. That "shouldn't" be happening at this point as literally the generalization of the machine and the os being installed has already been completed, and this is being done on a fresh windows template.

This situation happens occasionally right at the point of using salt-cloud to deploy the salt-minion installer with the following default configuration on an empty template (#57594). As I understand it, the pki key is created and installed by salt-cloud only once.

```
                master: {{ pillar["local"]["ip4"] | yaml_dquote }}
                startup_states: highstate

                ipv6: false
                transport: zeromq

                use_superseded:
                  - module.run

                saltenv: bootstrap
                pillarenv: bootstrap
```

After deploying this configuration, the highstate of the "bootstrap" environments swap this configuration file for another which changes the environments to "base" and configures the etcd returner from #51346. After changing the environment the machine is restarted (in essence to restart the service to use the new configuration file). If there's an error, the machine is also restarted in order to retry the highstate. This logic has been the same since 2018.3, and the issue seemed to manifest during an update from 2019.2 to 3000.3.

In this situation, (as I understand it) the pki key shouldn't be changing as it's not being directly touched by any of the states, and the only thing that's changing is the `saltenv` and `pillarenv` in the configuration. 

I get that it doesn't make much sense, but is there some other condition perhaps that would result in the salt-master denying the key after it already being accepted?

If there isn't some additional condition, I can dig more through things to get a better repro..and better logs. I'd just thought that this situation might be related to some hardening of what might be considered a legitimate minion as a result of the recent CVEs that were patched.

**Comment by arizvisa** [comment]
After upgrading to 3001, it seems this issue has gone away. I'm not sure which PR or what change fixed this, as I tried downgrading back to 3000.3 to confirm my sanity and was still able to reproduce.

So, closing this issue as you guys have resolved it somewhere in-between 3000.3 and 3001.
